components/ruby/lib/chef-licensing/license_key_fetcher.rb

```diff
@@ -83,9 +83,10 @@ def perform_on_prem_operations
         end
       end
 
-      # Scenario: When a user is prompted for license expiry and license is not yet renewed
-      if %i{prompt_license_about_to_expire prompt_license_expired_local_mode}.include?(config[:start_interaction])
-        # Not blocking any license type in case of expiry
+      # Scenario: When a user is prompted with license about to expire message and license is not yet renewed
+      # Scenario: When a user is prompted with license expired message in grace period and license is not yet renewed
+      if license && !license.expired?
```

Comment:
This will be true even for the best case, I assume; i.e. license is valid and not expired.

Reply:
License active check is done before and if it is active it is returned directly. If not then we prompt them first and then later this check again is done to decide if after the prompt we need to block or not based on license status.

Reply:
That makes sense now. Thanks